services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_utils.py

```diff
@@ -0,0 +1,371 @@
+
+    try:
+        last_task = await _extract_task_data_from_service_for_state(
+            service_id=service_id,
+            dynamic_sidecar_settings=dynamic_sidecar_settings,
+            target_statuses=TASK_STATES_ALL,
+        )
+    # GenericDockerError
+    except GenericDockerError as e:
+        if e.original_exception.message != f"service {service_id} not found":
+            raise e
+
+        # because the service is not there yet return a pending state
+        # it is looking for a service or something with no error message
+        pending_task_state = {"State": ServiceState.PENDING.value}
+        return extract_task_state(task_status=pending_task_state)
+
+    task_status = last_task["Status"]
+    return extract_task_state(task_status=task_status)
+
+
+async def are_services_missing(
+    node_uuid: str, dynamic_sidecar_settings: DynamicSidecarSettings
+) -> bool:
+    """Used to check if the service should be created"""
+    async with docker_client() as client:  # pylint: disable=not-async-context-manager
+        stack_services = await client.services.list(
+            filters={
+                "label": [
+                    f"swarm_stack_name={dynamic_sidecar_settings.swarm_stack_name}",
+                    f"uuid={node_uuid}",
+                ]
+            }
+        )
+        return len(stack_services) == 0
+
+
+async def are_all_services_present(
+    node_uuid: str, dynamic_sidecar_settings: DynamicSidecarSettings
+) -> bool:
+    """
+    The dynamic-sidecar stack always expects to have 2 running services
+    """
+    async with docker_client() as client:  # pylint: disable=not-async-context-manager
+        stack_services = await client.services.list(
+            filters={
+                "label": [
+                    f"swarm_stack_name={dynamic_sidecar_settings.swarm_stack_name}",
+                    f"uuid={node_uuid}",
+                ]
+            }
+        )
+        if len(stack_services) != 2:
+            log.warning("Expected 2 services found %s", stack_services)
+            return False
+
+        return True
+
+
+async def remove_dynamic_sidecar_stack(
+    node_uuid: str, dynamic_sidecar_settings: DynamicSidecarSettings
+) -> None:
+    """Removes all services from the stack, in theory there should only be 2 services"""
+    async with docker_client() as client:  # pylint: disable=not-async-context-manager
+        services_to_remove = await client.services.list(
+            filters={
+                "label": [
+                    f"swarm_stack_name={dynamic_sidecar_settings.swarm_stack_name}",
+                    f"uuid={node_uuid}",
+                ]
+            }
+        )
+        to_remove_tasks = [
+            client.services.delete(service["ID"]) for service in services_to_remove
+        ]
+        await asyncio.gather(*to_remove_tasks)
+
+
+async def remove_dynamic_sidecar_network(network_name: str):
+    with suppress(aiodocker.exceptions.DockerError):
```

Comment:
here you swallow the error. would it not make sense to have it in the logs?

Reply:
Sometimes, there is an error when removing the network. While the dynamic-sidecar spawned containers (which are using this network) are still being killed, the error will be raised. Network removal is a bit strange in Docker.

Logging the error might be a bit confusing. I'm also a bit confused about the network removal, not a very polished thing, but It seems that the networks is cleaned up in the end.

Reply:
So... let me put it on warning. With some explanation for this error. Let's avoid coding Java style